erpnext/controllers/selling_controller.py

```diff
@@ -788,6 +789,10 @@ def validate_items(self):
 
 		validate_item_type(self, "is_sales_item", "sales")
 
+		for item in self.items:
+			if not item.product_bundle_name and self.has_product_bundle(item.item_code):
+				item.product_bundle_name = get_product_bundle(item.item_code).name
```

Comment:
fix: Always check if item is bundle, if not (after an update), clear the `product_bundle_name` field. Check that the Bundle named `product_bundle_name` is related to the item `item_code`